packages/mui-material/src/FilledInput/FilledInput.js

```diff
@@ -178,38 +213,59 @@ const FilledInputInput = styled(InputBaseInput, {
       },
     },
   }),
-  ...(ownerState.size === 'small' && {
-    paddingTop: 21,
-    paddingBottom: 4,
-  }),
-  ...(ownerState.hiddenLabel && {
-    paddingTop: 16,
-    paddingBottom: 17,
-  }),
-  ...(ownerState.startAdornment && {
-    paddingLeft: 0,
-  }),
-  ...(ownerState.endAdornment && {
-    paddingRight: 0,
-  }),
-  ...(ownerState.hiddenLabel &&
-    ownerState.size === 'small' && {
-      paddingTop: 8,
-      paddingBottom: 9,
-    }),
-  ...(ownerState.multiline && {
-    paddingTop: 0,
-    paddingBottom: 0,
-    paddingLeft: 0,
-    paddingRight: 0,
-  }),
+  variants: [
+    {
+      props: {
+        size: 'small',
+      },
+      style: {
+        paddingTop: 21,
+        paddingBottom: 4,
+      },
+    },
+    {
+      props: ({ ownerState }) => ownerState.hiddenLabel,
+      style: {
+        paddingTop: 16,
+        paddingBottom: 17,
+      },
+    },
+    {
+      props: ({ ownerState }) => ownerState.startAdornment,
+      style: {
+        paddingLeft: 0,
+      },
+    },
+    {
+      props: ({ ownerState }) => ownerState.endAdornment,
+      style: {
+        paddingRight: 0,
+      },
+    },
+    {
+      props: ({ ownerState }) => ownerState.hiddenLabel && ownerState.size === 'small',
+      style: {
+        paddingTop: 8,
+        paddingBottom: 9,
+      },
+    },
+    {
+      props: ({ ownerState }) => ownerState.multiline,
+      style: {
+        paddingTop: 0,
+        paddingBottom: 0,
+        paddingLeft: 0,
+        paddingRight: 0,
+      },
+    },
+  ],
 }));
 
 const FilledInput = React.forwardRef(function FilledInput(inProps, ref) {
   const props = useThemeProps({ props: inProps, name: 'MuiFilledInput' });
 
   const {
-    disableUnderline,
+    disableUnderline = false,
```

Comment:
This was needed because `undefined` failed somewhere? Or just to implement it properly?

I agree with the change, just wondering if we should have some eslint to avoid this in the future.

Reply:
The `variants` has this matching condition:

```js
variants: [ { props: { disableUnderline: false }, style: {…} } ],
```

without an explicit `false`, it will not match because `disableUnderline=undefined`